drivers/modem/modem_cellular.c

```diff
@@ -2109,44 +2109,54 @@ MODEM_CHAT_SCRIPT_DEFINE(sqn_gm02s_periodic_chat_script,
 #define MODEM_CELLULAR_GET_USER_PIPES(inst) \
 	MODEM_CELLULAR_INST_NAME(user_pipes, inst)
 
-#define MODEM_CELLULAR_DEVICE_QUECTEL_BG95(inst)                                                   \
-	MODEM_PPP_DEFINE(MODEM_CELLULAR_INST_NAME(ppp, inst), NULL, 98, 1500, 64);                 \
-                                                                                                   \
-	static struct modem_cellular_data MODEM_CELLULAR_INST_NAME(data, inst) = {                 \
-		.chat_delimiter = "\r",                                                            \
-		.chat_filter = "\n",                                                               \
-		.ppp = &MODEM_CELLULAR_INST_NAME(ppp, inst),                                       \
-	};                                                                                         \
-                                                                                                   \
-	MODEM_CELLULAR_DEFINE_USER_PIPE_DATA(                                                      \
-		inst,                                                                              \
-		user_pipe_0,                                                                       \
-		CONFIG_MODEM_CELLULAR_USER_PIPE_BUFFER_SIZES                                       \
-	);                                                                                         \
-                                                                                                   \
-	MODEM_CELLULAR_DEFINE_USER_PIPE_DATA(                                                      \
-		inst,                                                                              \
-		user_pipe_1,                                                                       \
-		CONFIG_MODEM_CELLULAR_USER_PIPE_BUFFER_SIZES                                       \
-	);                                                                                         \
-                                                                                                   \
+/* Extract the first argument (pipe name) from a pair */
+#define MODEM_CELLULAR_GET_PIPE_NAME_ARG(arg1, ...) arg1
+
+/* Extract the second argument (DLCI address) from a pair */
+#define MODEM_CELLULAR_GET_DLCI_ADDRESS_ARG(arg1, arg2, ...) arg2
+
+/* Define user pipe data using instance and extracted pipe name */
+#define MODEM_CELLULAR_DEFINE_USER_PIPE_DATA_HELPER(_args, inst)                                   \
```

Comment:
In this and the `MODEM_CELLULAR_INIT_USER_PIPE_HELPER()` macro is `_args` really more than just one argument? From reading `FOR_EACH_FIXED_ARG()` I wouldn't think so.

Reply:
`FOR_EACH_FIXED_ARG` passes one argument per iteration, but we use a trick with passing `(arg1, arg2)` and treat it as single argument.
That's why we need `MODEM_CELLULAR_GET_PIPE_NAME_ARG` and `MODEM_CELLULAR_GET_DLCI_ADDRESS_ARG`